engine-tests/src/utils/mod.rs

```diff
@@ -1000,11 +1000,15 @@ pub fn unwrap_revert_slice(result: &SubmitResult) -> &[u8] {
     }
 }
 
-pub fn panic_on_fail(status: TransactionStatus) {
+pub fn panic_on_fail(status: TransactionStatus) -> Option<String> {
```

Comment:
Maybe `get_error_message` is a better name for this function now that it no longer panics.